core/equipment.h

```diff
@@ -33,6 +33,17 @@ typedef struct
 	bool bestmix_he;
 } cylinder_t;
 
+/* Table of cylinders. Attention: this stores cylinders,
+ * *not* pointers to cylinders. This has two crucial consequences:
+ * 1) Pointers to cylinders are not stable. They may be
+ *    invalidated if the table is reallocated.
```

Comment:
This comment makes me worry: As far as I can see, this is currently not thread save, there are no locks. Even if it does not bite us now, it could at any point in the future introduce very subtle bugs. With this, it seems we should get rid of all cylinder_t pointers in the rest of the code (in favour of passing around indices) and also the construct get_cylinder(index)->member looks dangerous. Furthermore, if the table can grow and shrink (can it actually shrink? maybe that's not a good idea) at random times, keeping nr around for any delayed use can lead to problems. This clearly is not a full review, just a thing that popped to my mind when reading the code.